doc/releases/release-notes-4.2.rst

```diff
@@ -524,7 +524,7 @@ New Boards
 
 * Ai-Thinker Co., Ltd.
 
-   * :zephyr:board:`ai_wb2_12f` (``ai_wb2_12f``)
+   * :zephyr:board:`ai_wb2_12f_kit` (``ai_wb2_12f_kit``)
```

Comment:
Old board names will still be used in the old versions of the doc, i.e. [4.2 doc](https://docs.zephyrproject.org/4.3.0/boards/aithinker/ai_m62_12f/doc/index.html), all good.

This modification is about fixing  the dead-link in the new version of the doc, all good.